Added .ConfigureAwait(false) to every awaited async call

Leoltron opened the pull request:
https://github.com/Actify-Inc/arangodb-net-standard/issues/215

Leoltron commented:
I may overdid it a bit - you probably dont need `.ConfigureAwait(false)` in tests

DiscoPYF commented:
Thank you @Leoltron for this contribution. This is a good way to ensure the library will behave correctly when used in various asynchronous/synchronous ways.

From what I understand you created a tool to do this? I noticed this is very similar to other contributions you've made elsewhere, e.g. https://github.com/cmxl/FFmpeg.NET/pull/52

> I may overdid it a bit - you probably dont need .ConfigureAwait(false) in tests

I agree. I think we should remove the calls to `.ConfigureAwait(false)` in tests as they are not required and imo they add cluter. Some tests frameworks provide a synchronisation context when running the tests, this is the case of XUnit so we don't need to `.ConfigureAwait(false)` in the tests. We can consider the tests to be app-level code.

See also https://devblogs.microsoft.com/dotnet/configureawait-faq/

> So if you’re writing an event handler in a Windows Forms app, writing a unit test in xunit, writing code in an ASP.NET MVC controller, whether or not the app model did in fact publish a SynchronizationContext, you want to use that SynchronizationContext if it exists. And that means the default / ConfigureAwait(true). You make simple use of await, and the right things happen with regards to callbacks/continuations being posted back to the original context if one existed. This leads to the general guidance of: if you’re writing app-level code, do not use ConfigureAwait(false).

DiscoPYF commented:
There are now additional methods in `TransactionApiClient` to support stream tansactions, we should also add `.ConfigureAwait(false)` in them.

@Leoltron Are you likely to pick up the update? If not, we can add an additional commit on top of your contribution.

Leoltron commented:
Yes, its my first attempt to do something useful with Roslyn. I may eventually shape it into something I wont be ashamed to share with others.

Added `.ConfigureAwait(false)` to awaited async calls in TransactionApiClient, removed `.ConfigureAwait(false)` from tests.

DiscoPYF commented on arangodb-net-standard.Test/CollectionApi/CollectionApiClientTest.cs:
```diff
@@ -1,4 +1,4 @@
-﻿using ArangoDBNetStandard;
```
There is a change like this on the first line of each of the modified files. It's kinda weird. I can't see what's the change about. Maybe a line return thing?

Do you mind reverting those?

Leoltron replied:
Its UTF-8 byte order mark, Roslyn appears to save source files without it. Added it back to every source file.

Leoltron replied:
Now I might added it to files that didnt have it before, but at least now every file in repo has exactly the same encoding.

DiscoPYF replied:
Thanks 👍

DiscoPYF reviewed (commented):
Thanks for the update @Leoltron . There are a bunch of redundant changes at the top the files. I would like to double check what they are for before merging. Could you revert them if convenient? Then we can merge this directly

DiscoPYF commented:
Looks good to me, approved. Thanks for the contribution. 🥳 Merging now.
